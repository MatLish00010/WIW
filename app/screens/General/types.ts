import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { GeneralStackParamList } from '@myapp/navigation/GeneralStack/types';

export type GeneralScreenProps = NativeStackScreenProps<GeneralStackParamList>;
