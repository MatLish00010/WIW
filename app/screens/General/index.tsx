import { FC } from 'react';

import { GeneralProps } from '@myapp/navigation/GeneralStack/types';
import { ViewContainer, Button, ButtonText } from '@myapp/ui';

const General: FC<GeneralProps<'General'>> = ({ navigation }) => {
  return (
    <ViewContainer testID="general-screen">
      <Button testID="menu-btn" onPress={() => navigation.navigate('MenuStackNavigator', { screen: 'Kitchens' })}>
        <ButtonText>Menu</ButtonText>
      </Button>
    </ViewContainer>
  );
};

export default General;
