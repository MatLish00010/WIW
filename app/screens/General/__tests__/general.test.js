import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react-native';

import General from '@myapp/screens/General';
import testingWrapper from '@myapp/utils/tests/tistingWrapper';

describe('General Screen Test', () => {
  test('Screen is correct', () => {
    const tree = renderer.create(testingWrapper(<General />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Text in buttons', () => {
    render(testingWrapper(<General />));
    expect(screen.getByTestId('menu-btn')).toHaveTextContent(/Menu/i);
  });
});
