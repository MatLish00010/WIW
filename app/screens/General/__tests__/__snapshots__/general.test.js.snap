// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Screen Test Screen is correct 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
        "paddingBottom": 32,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 40,
      },
    ]
  }
  testID="general-screen"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2f95dc",
        "borderRadius": 8,
        "display": "flex",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 16,
      }
    }
    testID="menu-btn"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
          },
        ]
      }
    >
      Menu
    </Text>
  </View>
</View>
`;

exports[`Home Screen Test Screen is correct 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
        "paddingBottom": 32,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 40,
      },
    ]
  }
  testID="general-screen"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2f95dc",
        "borderRadius": 8,
        "display": "flex",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 16,
      }
    }
    testID="menu-btn"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
          },
        ]
      }
    >
      Menu
    </Text>
  </View>
</View>
`;
